python/ray/train/sklearn/sklearn_predictor.py

```diff
@@ -23,16 +23,27 @@ class SklearnPredictor(Predictor):
             predictions.
         preprocessor: A preprocessor used to transform data batches prior
             to prediction.
+        use_gpu: If set, the model will be moved to GPU on instantiation and
```

Comment:
Hmm should we not require `use_gpu` as part of the base Predictor API? Each Predictor can choose to expose it if it's supported. We currently don't have this arg for XGBoostPredictor, HuggingfacePredictor, etc.

For the `BatchPredictor` maybe we can wrap the automatic arg setting in a try-except and only set it if the Predictor contains the arg.

Reply:
i changed to we only set use_gpu flag if user explicitly passed in gpu resources so sklearn_predictor doesn't need to expose it